File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const answer = [];
    const N = name.length;
    const map = new Map();
    
    for (let i = 0; i < N; i++) {
        map.set(name[i], yearning[i]);
    }
    
    for (let p of photo) {
        let score = 0;
        
        for (let n of p) {
            score += map.get(n) || 0; 
        }
        
        answer.push(score);
    }
    
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    score_map = { x: yearning[i] for (i, x) in enumerate(name) }
    result = []
    
    for ph in photo:
        score = sum(map(lambda x: score_map[x] if x in score_map else 0, ph))
        result.append(score)
        
    return result